content/testing.md

```diff
@@ -0,0 +1,256 @@
+---
+title: "Testing"
+---
+
+**NOTE** This is correct up to release 1.3-beta.12. Apologies that this is still a work in progress.
+
+<h2 id="testing-applications">Testing your Application</h2>
+
+There are many benefits of testing your application to ensure it works the way you think it does. Reasons include maintaining a high level of quality (especially over time as your codebase changes), allowing you to refactor and rewrite code with confidence, and concrete documentation of expected behavior. (Other developers can figure out what parts of your app are supposed to do by reading the tests!)
+
+Automated testing allows you to do all of these things to a much greater degree and *run tests more often*, which means your codebase will remain in better shape and regress less. 
+
+<h3 id="testing-concepts">Testing concepts</h3>
+
+Entire books have been written on the subject of testing, so we will simply touch on some basics of testing here. The important thing to consider when writing a test is what part of the application you are trying to test, and how you are verifying the behaviour works.
+
+If you are testing one small module of your application, you are writing a *unit test*. You'll need to take steps to *stub* and *mock* other modules that your module usually leverages in order to *isolate* each test. You'll typically also need to *spy* on actions that the module takes to verify that they occur.
```

Comment:
I think this should be a bulleted list with the name of the testing type in bold at the beginning. From here it's not clear that this is an emumeration of types of testing.